projects/canopy/src/lib/spacing/padding/padding.directive.ts

```diff
@@ -14,69 +14,78 @@ import { SpacingVariant } from '../spacing.interface';
-      `lg-padding__top--${padding}`,
-      this.paddingTopClass,
-    );
+  set lgPaddingTop(padding: SpacingVariant | ResponsiveSpacing) {
+    const newClasses = this.createNewClasses(padding, 'lg-padding__top');
+    this.paddingTopClasses = this.toggleClasses(newClasses, this.paddingTopClasses);
   }
 
-  paddingRightClass: string;
+  paddingRightClasses: Array<string> = [];
   @Input()
-  set lgPaddingRight(padding: SpacingVariant) {
-    this.paddingRightClass = this.togglePaddingClass(
-      `lg-padding__right--${padding}`,
-      this.paddingRightClass,
-    );
+  set lgPaddingRight(padding: SpacingVariant | ResponsiveSpacing) {
+    const newClasses = this.createNewClasses(padding, 'lg-padding__right');
+    this.paddingRightClasses = this.toggleClasses(newClasses, this.paddingRightClasses);
   }
 
-  paddingBottomClass: string;
+  paddingBottomClasses: Array<string> = [];
   @Input()
-  set lgPaddingBottom(padding: SpacingVariant) {
-    this.paddingBottomClass = this.togglePaddingClass(
-      `lg-padding__bottom--${padding}`,
-      this.paddingBottomClass,
-    );
+  set lgPaddingBottom(padding: SpacingVariant | ResponsiveSpacing) {
+    const newClasses = this.createNewClasses(padding, 'lg-padding__bottom');
+    this.paddingBottomClasses = this.toggleClasses(newClasses, this.paddingBottomClasses);
   }
 
-  paddingLeftClass: string;
+  paddingLeftClasses: Array<string> = [];
   @Input()
-  set lgPaddingLeft(padding: SpacingVariant) {
-    this.paddingLeftClass = this.togglePaddingClass(
-      `lg-padding__left--${padding}`,
-      this.paddingLeftClass,
-    );
+  set lgPaddingLeft(padding: SpacingVariant | ResponsiveSpacing) {
+    const newClasses = this.createNewClasses(padding, 'lg-padding__left');
+    this.paddingLeftClasses = this.toggleClasses(newClasses, this.paddingLeftClasses);
   }
 
   @Input()
-  set lgPaddingHorizontal(padding: SpacingVariant) {
+  set lgPaddingHorizontal(padding: SpacingVariant | ResponsiveSpacing) {
     this.lgPaddingLeft = padding;
     this.lgPaddingRight = padding;
   }
 
   @Input()
-  set lgPaddingVertical(padding: SpacingVariant) {
+  set lgPaddingVertical(padding: SpacingVariant | ResponsiveSpacing) {
     this.lgPaddingTop = padding;
     this.lgPaddingBottom = padding;
   }
 
-  paddingClass: string;
+  paddingClasses: Array<string> = [];
   @Input()
-  set lgPadding(padding: SpacingVariant) {
-    this.paddingClass = this.togglePaddingClass(
-      `lg-padding--${padding}`,
-      this.paddingClass,
-    );
+  set lgPadding(padding: SpacingVariant | ResponsiveSpacing) {
+    const newClasses = this.createNewClasses(padding, 'lg-padding');
+    this.paddingClasses = this.toggleClasses(newClasses, this.paddingClasses);
   }
 
-  togglePaddingClass(newClass: string, oldClass: string): string {
-    if (oldClass) {
-      this.renderer.removeClass(this.hostElement.nativeElement, oldClass);
+  createNewClasses(padding: SpacingVariant | ResponsiveSpacing, classPrefix: string) {
```

Comment:
There's something I don't quite like about this. I think it's the fact that `Spacing` is the first word on one, and the second on the other. I also wonder if `ResponsiveSpacing` should be more of a noun? e.g. `SpacingRules`. (not too sure)

Reply:
How about `SpacingResponsive`? I quite like it - short and to the point. I've added that for now in my most recent commit, but can change it again.

Reply:
Changed to `SpacingResponsive`